app/controllers/katello/concerns/api/v2/repository_content_controller.rb

```diff
@@ -105,9 +114,12 @@ def sort_options
         end
       end
 
-      def filter_by_content_view_filter(filter, collection)
-        ids = filter.send("#{singular_resource_name}_rules").pluck(:uuid)
-        filter_by_ids(ids, collection)
```

Comment:
This is only applicable to package groups (which uses uuid) and is failing for other content type queries. I moved this to raise `NotImplemented` errors if the child class doesn't implement the method and moved this implementation to package groups.

With this PR, this is overridden in the following child classes:
```
app/controllers/katello/api/v2/docker_manifests_controller.rb:    def filter_by_content_view_filter(filter)
app/controllers/katello/api/v2/errata_controller.rb:    def filter_by_content_view_filter(filter, collection)
app/controllers/katello/api/v2/module_streams_controller.rb:    def filter_by_content_view_filter(filter, collection)
app/controllers/katello/api/v2/package_groups_controller.rb:    def filter_by_content_view_filter(filter, collection)
app/controllers/katello/api/v2/packages_controller.rb:    def filter_by_content_view_filter(filter, collection)
```